protoc-gen-twirp/generator.go

```diff
@@ -711,29 +711,30 @@ func (t *twirp) generateUtils() {
 	t.P()
 
 	t.P(`// doProtobufRequest makes a Protobuf request to the remote Twirp service.`)
-	t.P(`func doProtobufRequest(ctx `, t.pkgs["context"], `.Context, client HTTPClient, hooks *`, t.pkgs["twirp"], `.ClientHooks, url string, in, out `, t.pkgs["proto"], `.Message) (err error) {`)
+	t.P(`func doProtobufRequest(ctx `, t.pkgs["context"], `.Context, client HTTPClient, hooks *`, t.pkgs["twirp"], `.ClientHooks, url string, in, out `, t.pkgs["proto"], `.Message) (returnCtx `, t.pkgs["context"], `.Context, err error) {`)
```

Comment:
I'm not sure why these return variables were named to begin with, since the code doesn't use a "naked return". Could we just make these unnamed returns instead and declare the error variable at the top?

Reply:
There is one defer function that sets `err`:
https://github.com/twitchtv/twirp/pull/251/files/b1c5f0de3fd57b7de3e15eda135698b4cfbeaf08#diff-a3a2b0741a88a528d33eb1f40bef6b04L742
```
t.P(`  defer func() {`)
t.P(`    cerr := resp.Body.Close()`)
t.P(`    if err == nil && cerr != nil {`)
t.P(`      err = wrapInternal(cerr, "failed to close response body")`)
t.P(`    }`)
t.P(`  }()`)
```

That is right now the only reason to have named returns. 
I could move the first line `returnCtx = ctx` into the defer. Unsure what is best.
I could also switch to a complete naked return style by changing all current error returns.

Reply:
Your test case looks great, thanks for adding that! Also I totally missed this defer block earlier, so thanks for pointing it out.

What do you think of just making the return signature `(_ context.Context, err error)`? This seems slightly more clear, since we're only relying on the named behavior for one of the values.

Reply:
Changed the signature and rebased the change onto the main PR commit.

Also fixed a minor thing in the test. retool requires handling of all error returns and I ignored the service test call result so I added a check for that so that retool is happy.